[CONS-6524] Standardize OpenShift examples (#24920)
DataDog/documentation: content/en/containers/kubernetes/distributions.md

In an EKS cluster, you can install the Operator using [Helm][1] or as an [EKS add-on][2].

> The configuration below is meant to work with either set up (Helm or EKS add-on) when the Agent is installed in the same namespace as the Datadog Operator.

Reviewer: ```suggestion
The configuration below is meant to work with either setup (Helm or EKS add-on) when the Agent is installed in the same namespace as the Datadog Operator.
```

Reviewer: (setup is the noun; set up is the verb)

{{% /tab %}}
{{< /tabs >}}

> Using `spec.nodeName` keeps TLS verification. In some clusters, DNS resolution for `spec.nodeName` inside Pods may not work in AKS. This has been reported on all AKS Windows nodes and when the cluster is set up in a Virtual Network using custom DNS on Linux nodes. In this case use the first AKS configuration provided: remove any settings for the Kubelet host path (defaults to `status.hostIP`) and use `tlsVerify: false`. This setting is **required**. Do NOT set the Kubelet host path and `tlsVerify: false` together.

Reviewer: ```suggestion
Using `spec.nodeName` keeps TLS verification. In some clusters, DNS resolution for `spec.nodeName` inside Pods may not work in AKS. This has been reported on all AKS Windows nodes, as well as Linux nodes when the cluster is set up in a Virtual Network using custom DNS. In this case, use the first AKS configuration provided: remove any settings for the Kubelet host path (which defaults to `status.hostIP`) and use `tlsVerify: false`. This setting is **required**. Do NOT set the Kubelet host path and `tlsVerify: false` in the same configuration.
```

## Red Hat OpenShift {#Openshift}

> OpenShift comes with hardened security by default with SELinux and SecurityContextConstraints (SCC), thus requiring some specific configurations:

Reviewer: ```suggestion
OpenShift comes with hardened security by default with SELinux and SecurityContextConstraints (SCC). As a result, it requires some specific configurations:
```

- Elevated SCC access for the Node Agent and Cluster Agent
- Kubelet API certificates may not always be signed by cluster CA
- Tolerations are required to schedule the Node Agent on `master` and `infra` nodes
- Cluster name should be set as it cannot be retrieved automatically from cloud provider

> - *(Optional)* Setting `hostNetwork: true` in the Node Agent to allow the Agent to make requests to cloud provider metadata services (IMDS)

Reviewer: ```suggestion
- *(Optional)* Set `hostNetwork: true` in the Node Agent to allow the Agent to make requests to cloud provider metadata services (IMDS)
```

> This core configuration supports OpenShift 3.11 and OpenShift 4, but works best with OpenShift 4.

Reviewer: ```suggestion
This core configuration supports OpenShift 3.11 and OpenShift 4, but it works best with OpenShift 4.
```

> Additionally log collection and APM have slightly different requirements as well.

Reviewer: ```suggestion
Log collection and APM also have specific requirements.
```

> The use of Unix Domain Socket (UDS) for APM and DogStatsD can work in OpenShift, however is inadvisable. As it requires additional privileged permissions and SCC access to **both** your Datadog Agent pod and your application pod. Without these your application pod can fail to deploy. Datadog recommends disabling the UDS option to avoid this, allowing the Admission Controller inject the appropriate [TCP/IP setting][7] or [Service setting][8] for APM connectivity.

Reviewer: ```suggestion
The use of Unix Domain Socket (UDS) for APM and DogStatsD can work in OpenShift. However, Datadog does not recommend this, as it requires additional privileged permissions and SCC access to **both** your Datadog Agent pod and your application pod. Without these, your application pod can fail to deploy. Datadog recommends disabling the UDS option to avoid this, allowing the Admission Controller to inject the appropriate [TCP/IP setting][7] or [Service setting][8] for APM connectivity.
```

{{< tabs >}}
{{% tab "Datadog Operator" %}}

> When using the Datadog Operator in OpenShift, it is recommended that you use the Operator Lifecycle Manager to deploy the Datadog Operator from OperatorHub in your OpenShift Cluster web console. You can find the [Operator install steps here][1].
> 
> The configuration below is meant to work with this set up which creates the [ClusterRole and ClusterRoleBinding based access to the SCC][2] for this specified ServiceAccount `datadog-agent-scc`. This `DatadogAgent` configuration is meant to be deployed in the same namespace as the Datadog Operator.

Reviewer: ```suggestion
When using the Datadog Operator in OpenShift, Datadog recommends that you use the Operator Lifecycle Manager to deploy the Datadog Operator from OperatorHub in your OpenShift Cluster web console. Refer to the [Operator install steps][1]. The configuration below works with that setup, which creates the [ClusterRole and ClusterRoleBinding based access to the SCC][2] for the specified ServiceAccount `datadog-agent-scc`. This `DatadogAgent` configuration should be deployed in the same namespace as the Datadog Operator.
```

```yaml
kind: DatadogAgent
apiVersion: datadoghq.com/v2alpha1
metadata:
  name: datadog

>   namespace: openshift-operators # set as namespace as where the Datadog Operator was deployed

Reviewer: ```suggestion
  namespace: openshift-operators # set as the same namespace where the Datadog Operator was deployed
```
